lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -356,3 +380,69 @@ def test_filter_query(self, monkeypatch, filters, expected):
             )
             == prefix + expected
         )
+
+    def test_user_no_auth(self, monkeypatch, db_session):
+        """Test the authorization error when a match against a key in the user
+        namespace is attempted from an unauthenticated session.
+        """
+        monkeypatch.setattr(
+            "pbench.server.api.resources.datasets_list.Auth.get_current_user_id",
+            lambda: None,
+        )
+        with pytest.raises(APIAbort) as e:
+            DatasetsList.filter_query(
+                ["user.foo:1"], Database.db_session.query(Dataset).outerjoin(Metadata)
+            )
+        assert e.value.http_status == HTTPStatus.UNAUTHORIZED
```

Comment:
This test looks redundant given the parametrization at line 401.

Reply:
Yeah, and I'd meant to remove this but I forgot. At this point, a redundant test won't hurt and I'd rather get it in than deal with another round.